src/rewriters/Rewritings.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2024 Martin Blicha <martin.blicha@gmail.com>
+ *
+ *  SPDX-License-Identifier: MIT
+ *
+ */
+
+#ifndef OPENSMT_REWRITINGS_H
+#define OPENSMT_REWRITINGS_H
```

Comment:
I understand the idea of distinguishing the files with classes and free functions, but I am afraid that it is a bit confusing.
I mean: Since `Rewriter.h` and `Rewritings.h` sound almost the same, it may be confusing which file the developer should choose and it may not be clear what the difference between them is.

Maybe adding `#include "Rewriter.h"` here could help to see that this is an extension and not a completely different thing?
Or to avoid unnecessary dependency, at least provide a short comment?